pkgs/applications/audio/aacgain/default.nix

```diff
@@ -1,27 +1,26 @@
 { lib
 , stdenv
-, fetchFromGitHub
+, fetchgit
 , cmake
 , autoconf
 , automake
 , libtool
-, faad2
 , mp4v2
 }:
 
 stdenv.mkDerivation rec {
   pname = "aacgain";
   version = "2.0.0";
 
-  src = fetchFromGitHub {
-    owner = "dgilman";
-    repo = pname;
+  src = fetchgit {
+    url = "https://github.com/dgilman/aacgain.git";
+    name = pname;
     rev = version;
-    sha256 = "sha256-9Y23Zh7q3oB4ha17Fpm1Hu2+wtQOA1llj6WDUAO2ARU=";
+    hash = "sha256-842+ueBSrTRs/e14d2LUd+uFi2qgJOYv+dswpC0lgIo=";
   };
 
   postPatch = ''
-    cp -R ${faad2.src}/* 3rdparty/faad2
+    rm -rf 3rdparty/mp4v2/*
```

Comment:
That was what Sandro discouraged in https://github.com/NixOS/nixpkgs/pull/203130#discussion_r1038511261, but I guess we don't have any other simple choices...

Reply:
Yes, from a "I want to get this package working" perspective, I don't really see any good approaches here. What came to my mind:

- This hybrid approach here
- Use the nixpkgs version of `faad2`, and downgrade it through overrides (to me, this sound like the same thing, but way more complicated)
- Switch entirely to the vendored 3rd party libs, as that's what the upstream project is effectively built against
- Somehow patch the CMake setup to work properly with the current nixpkgs version of faad2 (complicated, and frankly wrestling with CMake is beyond my pain tolerance threshold)

Reply:
I'd prefer to switch entirely to the vendored 3rd party libs. Who knows if a future `mp4v2` update won't break it again...

Reply:
FWIW: i did add the patches needed to build with the updated faad2 package https://github.com/NixOS/nixpkgs/compare/master...a-n-n-a-l-e-e:nixpkgs:aacgain-updated-libs but it seemed like an unnecessary maintenance burden with zero upside as the libs must be statically linked so aacgain can use interfaces that are not exported in the shared libs.